qark/plugins/broadcast/send_broadcast_receiver_permission.py

```diff
@@ -0,0 +1,196 @@
+
+
+log = logging.getLogger(__name__)
+
+BROADCAST_METHODS = (
+    "sendBroadcast",
+    "sendBroadcastAsUser",
+    "sendOrderedBroadcast",
+    "sendOrderedBroadcastAsUser",
+    "sendStickyBroadcast",
+    "sendStickyBroadcastAsUser",
+    "sendStickyOrderedBroadcast",
+    "sendStickyOrderedBroadcastAsUser",
+)
+STICKY_BROADCAST_METHODS = (
+    "sendStickyBroadcast",
+    "sendStickyBroadcastAsUser",
+    "sendStickyOrderedBroadcast",
+    "sendStickyOrderedBroadcastAsUser",
+)
+LOCAL_BROADCAST_IMPORTS = (
+    "android.support.v4.content.LocalBroadcastManager",
+    "android.support.v4.content.*",
+    "android.support.v4.*",
+    "android.support.*",
+    "android.*",
+)
+BROADCAST_WITHOUT_RECEIVER = (
+    "A broadcast, {broadcast_type}, is sent from this file {file_name}, which does not specify the receiverPermission. "
+    "This means any application on the device can receive this broadcast. You should investigate this "
+    "for potential data leakage."
+)
+BROADCAST_WITH_RECEIVER = (
+    "A broadcast, {broadcast_type}, is sent from this file {file_name}, which specifies the receiverPermission, "
+    "but depending on the protection level of the permission (on the receiving app side), may "
+    "still be vulnerable to interception, if the protection level of the permission is not set "
+    "to signature or signatureOrSystem. You should investigate this for potential data leakage."
+)
+BROADCAST_WITH_RECEIVER_UNDER_21 = (
+    "A broadcast, {broadcast_type}, is sent from this class: {file_name}, which specifies the "
+    "receiverPermission, but may still be vulnerable to interception, due to the permission squatting vulnerability "
+    "in API levels before 21. This means any application, installed prior to the expected receiver(s) on the device "
+    "can potentially receive this broadcast. You should investigate this for potential data leakage."
+)
+STICKY_BROADCAST = (
+    "A sticky broadcast, {broadcast_type}, is sent from this class: {file_name}. These should not be used, as they "
+    "rovide no security (anyone can access them), no protection (anyone can modify them), and many other problems. "
+    "For more info: http://developer.android.com/reference/android/content/Context.html"
+)
+
+
+class SendBroadcastReceiverPermission(BasePlugin):
+    """
+    This plugin checks certain broadcast methods to see if they are using an insecure version,
+    based on number of arguments.
+    """
+    def __init__(self):
+        BasePlugin.__init__(self, category="broadcast")
+        self.severity = Severity.WARNING
+        self.current_file = None
+        self.manifest_xml = None
+        self.below_min_sdk_21 = False
+
+    def run(self, files, apk_constants=None):
+        try:
+            self.below_min_sdk_21 = apk_constants["min_sdk"] < 21
+        except (KeyError, TypeError):
+            for decompiled_file in files:
+                if decompiled_file.lower().endswith("androidmanifest.xml"):
```

Comment:
Couldn't we just filter for androidmanifest.xml? Would it start with something else?

Reply:
`decompiled_file` is going to be the absolute path for these files. The thing we can do is to see if it ends with `/AndroidManifest.xml` or `\AndroidManifest.xml`, those approaches are a bit hacky but I don't know of another way. I'm open to ideas though.

Reply:
import ntpath
ntpath.basename(<path>)

Reply:
I've looked through the docs and think we can just use `os.sep` instead.